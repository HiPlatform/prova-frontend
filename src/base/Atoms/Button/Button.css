.uiButton {
  appearance: none;
  background-color: transparent;
  border: 0;
  text-align: left;
  color: #9658f4;
  cursor: pointer;
  font-size: 11px;
  font-weight: 600;
  outline: none;
  overflow: hidden;
  position: relative;
  text-decoration: none;
  width: 90%;
  padding: 2px 0px; }
  .uiButton.solid {
    background-color: #9658f4;
    text-align: center;
    color: #fff;
    font-weight: 600;
    padding: 10px 20px;
    width: 100%; }
