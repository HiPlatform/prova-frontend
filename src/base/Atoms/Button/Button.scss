@import "../../../variables";

.uiButton {
  appearance: none;
  background-color: transparent;
  border: 0;
  text-align: left;
  color: $primaryColor;
  cursor: pointer;
  font-size: 11px;
  font-weight: 600;
  outline: none;
  overflow: hidden;
  position: relative;
  text-decoration: none;
  width: 90%;
  padding: 2px 0px;

  &.solid {
    background-color: $primaryColor;
    text-align: center;
    color: $color;
    font-weight: 600;
    padding: 10px 20px;
    width: 100%;
  }
}