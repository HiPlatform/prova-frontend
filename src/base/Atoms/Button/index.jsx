import React from 'react';
import PropTypes from 'prop-types';
import Ink from 'react-ink';

import './Button.css';

const Button = ({ children, onClick }) => (
  <button className="uiButton" onClick={onClick}>
    {children}
    <Ink />
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};

export default Button;
