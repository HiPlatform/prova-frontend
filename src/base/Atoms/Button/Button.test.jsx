import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('Button', () => {
  const onClick = jest.fn();
  const props = {
    children: 'CLICK ME',
    onClick,
  };

  const component = shallow(<Button {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onClick function', () => {
    component.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should have the child prop rendered', () => {
    const childrenText = component.children()
      .reduce((text, child) => `${text}${child.text().replace('<e />', '')}`, '');
    expect(childrenText).toBe(props.children);
  });

  it('should render the modifier class', () => {
    const modifier = 'inverted';
    component.setProps({ modifier });
    expect(component.hasClass(modifier)).toBe(true);
  });
});
