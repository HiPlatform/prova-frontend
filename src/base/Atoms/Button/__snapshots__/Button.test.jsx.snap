// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    modifier=""
    onClick={[Function]}
>
    CLICK ME
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "CLICK ME",
        <e
          background={true}
          duration={1000}
          hasTouch={undefined}
          opacity={0.25}
          radius={150}
          recenter={true}
/>,
      ],
      "className": "uiButton ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "CLICK ME",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "background": true,
          "duration": 1000,
          "hasTouch": undefined,
          "opacity": 0.25,
          "radius": 150,
          "recenter": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "CLICK ME",
          <e
            background={true}
            duration={1000}
            hasTouch={undefined}
            opacity={0.25}
            radius={150}
            recenter={true}
/>,
        ],
        "className": "uiButton ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "CLICK ME",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": true,
            "duration": 1000,
            "hasTouch": undefined,
            "opacity": 0.25,
            "radius": 150,
            "recenter": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
